### PR TITLE
Feature/hierachical ordering

### PR DESCRIPTION
**Pull request by stummk**
Adds the folder view to app.

we-kode/mml.project#3

### CODE REVIEW

File: lib/models/model_base.dart [hunk 1]
```diff
@@ -5,11 +5,17 @@ abstract class ModelBase {
   /// Indicates whether the model object is deletable.
   late bool? isDeletable;
 
+  /// Indicates whether the model object is deletable.
```
Comment (dennis95stumm) [inline lib/models/model_base.dart:8]: selectable?

File: lib/view_models/records/record_tag_filter.dart [hunk 2]
```diff
@@ -11,18 +12,32 @@ class RecordTagFilterViewModel extends ChangeNotifier {
...
+  final SecureStorageService _storage = SecureStorageService.getInstance();
+
   /// Initializes the view model.
   RecordTagFilterViewModel(this.tagFilter);
 
   /// Clears the filter value of the [identifier].
   void clear(String identifier) {
     tagFilter.clear(identifier);
+     if (identifier == ID3TagFilters.folderView) {
+     _storage.set(
```
Comment (dennis95stumm) [inline lib/view_models/records/record_tag_filter.dart:24]: await?

Comment (dennis95stumm) [reply lib/view_models/records/record_tag_filter.dart:24]: Should here really the folder view be resetted

**Review by dennis95stumm — CHANGES_REQUESTED** [review]

File: lib/view_models/records/record_tag_filter.dart (hunk 2 above)
Comment (stummk) [reply lib/view_models/records/record_tag_filter.dart:24]: Yes.

**Review by dennis95stumm — APPROVED** [review]

**Pull request merged**